app/controllers/patients_controller.rb

```diff
@@ -1,9 +1,15 @@
 # Create and update patients, plus the main patient view in edit
 class PatientsController < ApplicationController
+  before_action :confirm_user_has_data_access, only: [:index]
   before_action :find_patient, only: [:edit, :update]
   rescue_from Mongoid::Errors::DocumentNotFound,
               with: -> { redirect_to root_path }
 
+  def index
+    @patients = Patient.all
+    respond_to :csv
```

Comment:
It's been a few rails versions since I've had to write this, but I think we can rewrite this line as follows to just have it download, using the `send_data` method (which is basically saying to rails 'download this instead of showing it in browser'):

```rb
respond_to do |format|
  format.csv { send_data @patients }
end
```